File: Together/Android/QR_Read/app/src/main/java/com/soyu/qr_read/MainActivity.java
package com.soyu.qr_read;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.google.zxing.integration.android.IntentIntegrator;
import com.google.zxing.integration.android.IntentResult;

public class MainActivity extends AppCompatActivity {
    private MainActivity context;

    public static int REQUEST_QR_READ_DATA = 49374;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        context = this;

        Button bt_qrread = (Button)findViewById(R.id.bt_qrread);
        bt_qrread.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                IntentIntegrator intentIntegrator = new IntentIntegrator(context);
                intentIntegrator.setBeepEnabled(false);//바코드 인식시 소리
                intentIntegrator.setCaptureActivity(ScanActivity.class);
                intentIntegrator.initiateScan();
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        Log.e("!!!", "requestCode :: "+requestCode );
        Log.e("!!!", "resultCode :: "+resultCode );
    if (requestCode == REQUEST_QR_READ_DATA) {
        IntentResult result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data);
        if(result != null && result.getContents() != null) {
            Log.e("!!!", "REQUEST_QR_READ_DATA :: "+result.getContents());
        }
    }
    }
}

File: Together/Kotlin/Together_Test/app/src/main/java/com/soyu/together_test/MainActivity.kt
package com.soyu.together_test

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import okhttp3.*
import okio.ByteString
import java.util.concurrent.TimeUnit

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bt_login = findViewById(R.id.bt_login) as Button
        bt_login.setOnClickListener {
            Log.e("!!!", "bt_login setOnClickListener")

            val client = OkHttpClient.Builder()
                .readTimeout(3, TimeUnit.SECONDS)
                //.sslSocketFactory() - ? нужно ли его указывать дополнительно
                .build()
            val request = Request.Builder()
                .url("ws://steeloz.ddns.net:4877") // 'wss' - для защищенного канала
                .build()
            val wsListener = EchoWebSocketListener ()
            val webSocket = client.newWebSocket(request, wsListener) // this provide to make 'Open ws connection'

        }

    }
}











private class EchoWebSocketListener : WebSocketListener() {
    override fun onOpen(webSocket: WebSocket, response: Response) {
        output("onOpen ")
        webSocket.send("Hello, it's SSaurel !")
        webSocket.send("What's up ?")
        webSocket.send(ByteString.decodeHex("deadbeef"))
        webSocket.close(NORMAL_CLOSURE_STATUS, "Goodbye !")
    }

    override fun onMessage(webSocket: WebSocket?, text: String?) {
        output("Receiving : " + text!!)
    }

    override fun onMessage(webSocket: WebSocket?, bytes: ByteString?) {
        output("Receiving bytes : " + bytes!!.hex())
    }

    override fun onClosing(webSocket: WebSocket?, code: Int, reason: String?) {
        webSocket!!.close(NORMAL_CLOSURE_STATUS, null)
        output("Closing : $code / $reason")
    }

    override fun onFailure(webSocket: WebSocket, t: Throwable, response: Response?) {
        output("Error : " + t.message)
    }

    companion object {
        private val NORMAL_CLOSURE_STATUS = 1000
    }

    private fun output(txt: String) {
        Log.e("WSS", txt)
    }
}